pytestqt/wait_signal.py

```diff
@@ -54,11 +57,77 @@ def _quit_loop_by_timeout(self):
             self._loop.quit()
 
     def _cleanup(self):
+        # store timeout message before the data to construct it is lost
+        self._timeout_message = self.get_timeout_error_message()
         if self._timer is not None:
             _silent_disconnect(self._timer.timeout, self._quit_loop_by_timeout)
             self._timer.stop()
             self._timer = None
 
+    def get_timeout_error_message(self):
```

Comment:
I'm not sure this should be part of the public API, so it should have a `_` prefix (It is OK for subclasses to access members prefixed with  `_`).